src/hotspot/share/opto/compile.hpp

```diff
@@ -827,18 +835,39 @@ class Compile : public Phase {
 
   const CompilationFailureInfo* first_failure_details() const { return _first_failure_details; }
 
+  bool failing(DEBUG_ONLY(bool no_stress_bailout = false)) {
```

Comment:
It's somehow difficult to read what `failing(false/true)` now exactly mean. When having `failing(true)`, don't we get the same behavior as if we call `failing_internal()`? If `failing_internal()` is false, then we would only return false because we are not entering
```
    if (StressBailout && !no_stress_bailout) {
      return fail_randomly();
    }
```
So, I'm wondering if we cannot just use `failing_internal()` instead of `failing(true)` and remove the parameter completely?

Reply:
Thanks for the suggestions! Updated the PR.